tests/check-docs.rs

```diff
@@ -0,0 +1,60 @@
+use regex::Regex;
+use std::{collections::BTreeMap, fmt::Write, path::PathBuf};
+use walkdir::WalkDir;
+
+fn mangle_comments(s: &str) -> String {
+    let result = s
+        .split('\n')
+        .map(str::trim)
+        .filter_map(|line| line.strip_prefix("/// "))
+        .filter(|line| !line.starts_with("# ")) // filter out code
+        .map(|line| if line.starts_with("```") { "" } else { line })
+        .collect::<Vec<_>>();
+    result.join("\n")
+}
+
+const PREFIX: &str = r#"// autogenerated. Please look at /tests/check-docs.rs
+
+#[allow(clippy::too_many_lines)]
+#[rustfmt::skip]
+fn explain(lint: &str) {
+    println!("{}", match lint {
+"#;
+
+const SUFFIX: &str = r#"        _ => "unknown lint",
+    });
+}
+"#;
+
+static CODE_FILE: &str = "src/docs.rs";
+
+#[test]
+fn gen_docs() {
```

Comment:
Should this be done in `cargo dev update_lints` rather than in a test?

Reply:
I moved this to `update_lints`.